lib/msf/base/serializer/readable_text.rb

```diff
@@ -555,7 +555,7 @@ def self.dump_options(mod, indent = '', missing = false)
         ])
 
     mod.options.sorted.each do |name, opt|
-      val = mod.datastore[name].nil? ? opt.default : mod.datastore[name]
+      val = mod.datastore[name]
```

Comment:
The `.nil?` check fails if the value is actually set to nil, which meant the `show options` command didn't reflect what was actually in the datastore

Tracing the code change, it looks like the `.nil?` was added after it failed a similar issue with `false` being handled correctly with `val = mod.datastore[name] || opt.default` 

https://github.com/rapid7/metasploit-framework/commit/0660880332029327a77ac7ae67f308af859a8f4d